packages/runtime/container-runtime/src/dataStoreContext.ts

```diff
@@ -247,14 +247,15 @@ export abstract class FluidDataStoreContext extends TypedEventEmitter<IFluidData
         const thisSummarizeInternal =
             async (fullTree: boolean, trackState: boolean) => this.summarizeInternal(fullTree, trackState);
 
-        // Add self route (empty string) to used routes in the summarizer node. If GC is enabled, the used routes will
-        // be updated as per the GC data.
         this.summarizerNode = createSummarizerNode(
             thisSummarizeInternal,
             async () => this.getGCDataInternal(),
             async () => this.getInitialGCSummaryDetails(),
-            [""] /* usedRoutes */,
         );
+
+        // Add self route (empty string) to used routes in the summarizer node. If GC is enabled, the used routes will
+        // be updated as per the GC data.
+        this.summarizerNode.updateUsedRoutes([""]);
```

Comment:
To be honest, I feel like I'm lacking context, which likely points to a need to comment code better.
For example, it's not clear from the code why route to self needs to be added.
And lifetime - why this has to happen on constructor, and not be only interesting action at summarization timeframe.

Reply:
Currently, while GC work in progress, there is an option to disable GC by setting `runGC` flag to false in `IContainerRuntimeOptions`. So, we need to make data stores and DDSs referenced by default so that they are not marked as unreferenced.
There are 2 ways of doing it - Either make the default usedRoutes in `SummarizerNodeWithGC` to have self-route or set it explicitly in data store / channel context creation. I chose the latter so that later we don't have to change `SummarizerNodeWithGC`. But now that I am thinking about it, that may be a clearer option. We can change the default to be unreferenced once we remove the flag to disable GC.